src/Modal.jsx

```diff
@@ -31,6 +36,28 @@ class Modal extends React.Component {
 		}
 	}
 
+	/**
+	 * @param {String} scrollPosition window scroll position
+	 * @returns {String} CSS value for setting modal margin-top
+	 */
+	getModalPosition(scrollPosition) {
+		const viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
+
+		if (scrollPosition < viewportHeight) {
+			return DEFAULT_MARGIN_TOP;
```

Comment:
In Storybook, the modal isn't "snapping" to the top because it has this default margin